### PR TITLE
Test: Add integration test for Shardingsphere-proxy

### LINKED ISSUE #9797: [Feature] [Testing] Add IT to cover ShardingSphere DistSQL usage
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

We are using DistSQL in Sharding-MySQL storage option, but DistSQL is a new tech stack, and not stable.

The stability concern is noticed from 5.1.2 to 5.2.x version bumping. We have successfully passed all tests in 5.1.2 but failed in all 5.2 dev and released versions.

So, in order to ease the ShardingSphere community verification, we could build an Integration Testing based on test container similar to ElasticSearch client test.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wankai123**
Test: Add integration test for Shardingsphere-proxy
Fix `ShardingTopologyQueryDAO.loadServiceRelationsDetectedAtServerSide` invoke backend miss parameter `serviceIds`.

- [X] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #9797.
- [X] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

### CODE REVIEW

**Review by wu-sheng — APPROVED** [review]

**Comment by wu-sheng** [comment]
@terrymanu @tristaZero This test case should be able to help you locate the compatibility issue we faced in the integration.
Once the ITShardingTest could pass on your new release, I think it would be easy for us to bring more compatible versions of ShardingSphere into our e2e testing.

Also, we tried to use ShardingSphere to provide a PostgreSQL cluster option, but fail due to the DistSQL doesn't work as expected and as before.

File: oap-server/server-storage-plugin/storage-shardingsphere-plugin/src/test/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/shardingsphere/ITShardingTest.java [hunk 1]
```diff
@@ -0,0 +1,728 @@
...
+    private ModuleManager moduleManager;
+    private MySQLTableInstaller mySQLTableInstaller;
+    private ShardingSphereTableInstaller installer;
+    private DurationWithinTTL durationWithinTTL = DurationWithinTTL.INSTANCE;
+    private final String countQuery = "SELECT COUNT(*) AS rc FROM ";
+
+    @Before
+    public void init() {
+        if (dsType.equals(DataSourceType.MYSQL)) {
+            starEnv("docker-compose-mysql.yml", 3306);
```
Comment (kezhenxu94) [inline oap-server/server-storage-plugin/storage-shardingsphere-plugin/src/test/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/shardingsphere/ITShardingTest.java:164]: Typo 
```suggestion
            startEnv("docker-compose-mysql.yml", 3306);
```

File: oap-server/server-storage-plugin/storage-shardingsphere-plugin/src/test/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/shardingsphere/ServiceCpmMetrics.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import org.apache.skywalking.oap.server.core.remote.grpc.proto.RemoteData;
+import org.apache.skywalking.oap.server.core.storage.annotation.Column;
+
+@Stream(
+    name = "service_cpm",
+    scopeId = 1,
+    builder = ServiceCpmMetricsBuilder.class,
+    processor = MetricsStreamProcessor.class
+)
+public class ServiceCpmMetrics extends CPMMetrics {
```
[the remark this answers is not in the document]
Comment (wu-sheng) [reply oap-server/server-storage-plugin/storage-shardingsphere-plugin/src/test/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/shardingsphere/ServiceCpmMetrics.java:35]: Yes, it is from decompiling.

[the remark this answers is not in the document]
Comment (wankai123) [reply oap-server/server-storage-plugin/storage-shardingsphere-plugin/src/test/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/shardingsphere/ServiceCpmMetrics.java:42]: polished

**Review by kezhenxu94 — APPROVED** [review]

**Comment by wu-sheng** [comment]
@kezhenxu94 @wankai123 ElasticSearch and Sharding ITs take a lots of time, should we consider separating them in the CI?
More ElasticSearch versions could be tests, and sharding pg is on our roadmap too.

Is this possible?

File: oap-server/server-storage-plugin/storage-shardingsphere-plugin/src/test/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/shardingsphere/ITShardingTest.java [hunk 3]
```diff
@@ -0,0 +1,732 @@
...
+    public static Collection<Object[]> versions() {
+        return Arrays.asList(new Object[][] {
+            {
+                "5.1.2",
+                DataSourceType.MYSQL
+            }
+        });
+    }
+
+    public static DockerComposeContainer<?> ENVIRONMENT;
```
Comment (kezhenxu94) [inline oap-server/server-storage-plugin/storage-shardingsphere-plugin/src/test/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/shardingsphere/ITShardingTest.java:142]: This field doesn't need to be `static`

File: oap-server/server-storage-plugin/storage-shardingsphere-plugin/src/test/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/shardingsphere/ITShardingTest.java [hunk 4]
```diff
@@ -0,0 +1,732 @@
...
+    }
+
+    @SneakyThrows
+    @Test
+    public void test() {
+        trafficTest();
+        metricsTest();
+        tagsTest();
+        recordsTest();
+        topologyTest();
```
Comment (kezhenxu94) [inline oap-server/server-storage-plugin/storage-shardingsphere-plugin/src/test/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/shardingsphere/ITShardingTest.java:283]: You can just annotate each of these methods with `@Test` and remove this `test()` method.

**Comment by kezhenxu94** [comment]
> @kezhenxu94 @wankai123 ElasticSearch and Sharding ITs take a lots of time, should we consider separating them in the CI? More ElasticSearch versions could be tests, and sharding pg is on our roadmap too.
> 
> Is this possible?

We can do that

File: oap-server/server-storage-plugin/storage-shardingsphere-plugin/src/test/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/shardingsphere/ITShardingTest.java (hunk 4 above)
Comment (kezhenxu94) [reply oap-server/server-storage-plugin/storage-shardingsphere-plugin/src/test/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/shardingsphere/ITShardingTest.java:283]: > You can just annotate each of these methods with `@Test` and remove this `test()` method. So when any of these tests failed, you can directly see the result of failed method

NVM, adding `@Test` to each method will cause the environment to start and stop again and again

**Comment by wu-sheng** [comment]
Let's separate at least Sharding test out of existing ITs, and provide a new parameter to expose it for new CI pipeline.

**Review by wu-sheng — APPROVED** [review]
LGTM. @kezhenxu94 We tried to bump up TestContainer to 1.17.3, but ElasticSearch case seems incompatible. When you have time, please take a look.

**Pull request merged**